.ci/install.sh

```diff
@@ -12,34 +28,44 @@ if [ ! -f "${TRAVIS_BUILD_DIR}/install/bin/luarocks" ]; then
         cmake . -DWITH_LUAJIT21=ON -DCMAKE_INSTALL_PREFIX="${TRAVIS_BUILD_DIR}/install"
         make install
     popd
+else
+    echo "Using cached luarocks."
 fi
 
-mkdir "${HOME}/.luarocks"
-cp "${TRAVIS_BUILD_DIR}/install/etc/luarocks/config.lua" "$HOME/.luarocks/config.lua"
-echo "wrap_bin_scripts = false" >> "$HOME/.luarocks/config.lua"
-travis_retry luarocks --local install luafilesystem
-# for verbose_print module
-travis_retry luarocks --local install ansicolors
-travis_retry luarocks --local install busted 2.0.rc12-1
-#- mv -f $HOME/.luarocks/bin/busted_bootstrap $HOME/.luarocks/bin/busted
-travis_retry luarocks --local install luacov
-# luasec doesn't automatically detect 64-bit libs
-travis_retry luarocks --local install luasec OPENSSL_LIBDIR=/usr/lib/x86_64-linux-gnu
-travis_retry luarocks --local install luacov-coveralls --server=http://rocks.moonscript.org/dev
-travis_retry luarocks --local install luacheck
-travis_retry luarocks --local install lanes  # for parallel luacheck
+if [ ! -d "${HOME}/.luarocks" ]; then
```

Comment:
I should've tagged this WIP although it's functional as is. :-) I was also thinking of adding a couple of simple checks for other dependencies along these lines:

```bash
$ LUAROCKS_VERSION=2.4.2
$ luarocks --version | grep -o '2.4.2'
2.4.2
$ [ "$LUAROCKS_VERSION" = "$(luarocks --version | grep -o "$LUAROCKS_VERSION")" ] && echo "w00t" || echo "welp"
w00t
$ LUAROCKS_VERSION=2.4.3
$ [ "$LUAROCKS_VERSION" = "$(luarocks --version | grep -o "$LUAROCKS_VERSION")" ] && echo "w00t" || echo "welp"
welp
```

You could do the same thing quite simply with luarocks without doing anything more complicated because it creates directories like `~/.luarocks/lib/luarocks/rocks/busted/2.0.rc12-1`.

```bash
$ BUSTED_VERSION="2.0.rc12-1"
$ [ -d "$HOME/.luarocks/lib/luarocks/rocks/busted/$BUSTED_VERSION" ] && echo "w00t" || echo "welp"
w00t
$ BUSTED_VERSION="2.1"
$ [ -d "$HOME/.luarocks/lib/luarocks/rocks/busted/$BUSTED_VERSION" ] && echo "w00t" || echo "welp"
welp
```

Anyway, caching the `.luarocks` directory makes no difference *unless* the luarocks server are timing out for some reason. All you win by not downloading and compiling (mostly luasec) is wasted again by the extra time spent dealing with cache.